atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -389,26 +406,45 @@ def do_worker_build(self, release, cluster_info, koji_upload_dir, task_id):
                 except OsbsException:
                     pass
 
+    def select_and_start_cluster(self, platform):
+        ''' Choose a cluster and start a build on it '''
+
+        config = get_config(self.workflow)
+        clusters = config.get_enabled_clusters_for_platform(platform)
+
+        if not clusters:
+            raise UnknownPlatformException('No clusters found for platform {}!'
+                                           .format(platform))
+
+        cluster_fails = Counter()
+        retry_at = {}
+        possible_clusters = None
+
+        while True:
+            if not clusters:
+                raise RuntimeError('Could not find appropriate cluster for worker build.')
+            possible_clusters = self.get_clusters(clusters, cluster_fails, platform, retry_at)
+            for cluster in possible_clusters or []:
+                try:
+                    self.do_worker_build(self.release, cluster, self.koji_upload_dir,
+                                         self.fs_task_id)
+                    return
+                except RuntimeError:
+                    cluster_fails[cluster.name] += 1
+                    retry_at[cluster.name] = datetime.datetime.now()\
```

Comment:
Nit: can backslash-line-continuation be avoided here? Might look nicer to enclose the expression in (), or rephrase it somehow.